File: scw-route-consumer/src/main/webapp/app/dao/entity/widget.js
function Widget (){
    this.id = "";
    this.code = "";
    this.name = "";
    this.containerId = "";
    this.type = "";
    this.sort = 0;
    this.tmpFlg = "true";
    this.attConfigs = new Array();
    this.elements = new Array();
};
Widget.getWidgetByType = function(type, containerId, widgetId) {
    if(type == "@text"){
        var widgetTextSample = new WidgetTextSample();
        return widgetTextSample.createObject(containerId, widgetId);
    }
    if(type == "@datepicker"){
        var widgetDateSample = new WidgetDateSample();
        return widgetDateSample.createObject(containerId, widgetId);
    }
    if(type == "@daterangepicker"){
        var widgetDateRangeSample = new WidgetDateRangeSample();
        return widgetDateRangeSample.createObject(containerId, widgetId);
    }
    if(type == "@select"){
        var widgetSelectSample = new WidgetSelectSample();
        return widgetSelectSample.createObject(containerId, widgetId);
    }
    if(type == "@radio"){
        var widgetRadioSample = new WidgetRadioSample();
        return widgetRadioSample.createObject(containerId, widgetId);
    }
    if(type == "@check"){
        var widgetCheckSample = new WidgetCheckSample();
        return widgetCheckSample.createObject(containerId, widgetId);
    }
}
Widget.getWidgetSample = function (type, widget) {
    if(type == "@text"){
        var widgetTextSample = new WidgetTextSample();
        widgetTextSample.currWidget = widget;
        return widgetTextSample;
    }
    if(type == "@datepicker"){
        var widgetDateSample = new WidgetDateSample();
        widgetDateSample.currWidget = widget;
        return widgetDateSample;
    }
    if(type == "@daterangepicker"){
        var widgetDateRangeSample = new WidgetDateRangeSample();
        widgetDateRangeSample.currWidget = widget;
        return widgetDateRangeSample;
    }
    if(type == "@select"){
        var widgetSelectSample = new WidgetSelectSample();
        widgetSelectSample.currWidget = widget;
        return widgetSelectSample;
    }
    if(type == "@radio"){
        var widgetRadioSample = new WidgetRadioSample();
        widgetRadioSample.currWidget = widget;
        return widgetRadioSample;
    }
    if(type == "@check"){
        var widgetCheckSample = new WidgetCheckSample();
        widgetCheckSample.currWidget = widget;
        return widgetCheckSample;
    }
}

Widget.prototype = {
    constructor: Widget,
    getId: function () {
        return this.id;
    },
    getCode: function () {
        return this.code;
    },
    getName: function () {
        return this.name;
    },
    getContainerId: function () {
        return this.containerId;
    },
    getType: function () {
        return this.type;
    },
    getSort: function () {
        return this.sort;
    },
    getTmpFlg: function(){
        return this.tmpFlg;
    },
    getAttConfigs: function () {
        return this.attConfigs;
    },
    getElements: function () {
        return this.elements;
    },
    addAttribute: function (attConfig) {
        if(this.attConfigs == null){
            this.attConfigs = new Array();
        }
        this.attConfigs[this.attConfigs.length] = attConfig;
    },
    addElement: function (element) {
        if(this.elements == null){
            this.elements = new Array();
        }
        this.elements[this.elements.length] = element;
    }
}

File: scw-provider-webeditor/src/main/java/com/github/wangdasong/scwproviderwebeditor/dao/entity/Widget.java
/********************************************************************
 *
 * Copyright 2016-2017 nsw All rights reserved
 *
 ********************************************************************/
package com.github.wangdasong.scwproviderwebeditor.dao.entity;


import com.github.wangdasong.scwbasecore.dao.entity.base.BaseEntity;

import java.util.List;

public class Widget extends BaseEntity {

    private String code;
    private String name;
    private String type;
    private int sort = 9999;
    private String containerId;
    private String tmpFlg = "false";
    private List<AttConfig> attConfigs;
    private List<Element> elements;


    public String getId() {
        return id;
    }

    public String getCode() {
        return code;
    }

    public String getName() {
        return name;
    }

    public String getType() {
        return type;
    }

    public String getContainerId() {
        return containerId;
    }

    public String getTmpFlg() {
        return tmpFlg;
    }




    public void setId(String id) {
        this.id = id;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setContainerId(String containerId) {
        this.containerId = containerId;
    }

    public void setTmpFlg(String tmpFlg) {
        this.tmpFlg = tmpFlg;
    }

	public List<AttConfig> getAttConfigs() {
		return attConfigs;
	}

	public void setAttConfigs(List<AttConfig> attConfigs) {
		this.attConfigs = attConfigs;
	}

	public List<Element> getElements() {
		return elements;
	}

	public void setElements(List<Element> elements) {
		this.elements = elements;
	}

	public int getSort() {
		return sort;
	}

	public void setSort(int sort) {
		this.sort = sort;
	}



}
